app/services/editions/db/IssueQueries.scala

```diff
@@ -107,21 +107,26 @@ trait IssueQueries extends Logging {
   }
 
   def listIssues(edition: Edition, dateFrom: LocalDate, dateTo: LocalDate) = DB readOnly { implicit session =>
-    sql"""
-    SELECT
-      id,
-      name,
-      issue_date,
-      timezone_id,
-      created_on,
-      created_by,
-      created_email,
-      launched_on,
-      launched_by,
-      launched_email
-    FROM edition_issues
-    WHERE issue_date BETWEEN $dateFrom AND $dateTo AND name = ${edition.entryName}
-    """.map(EditionsIssue.fromRow(_)).list().apply()
+    val maybeIssues = sql"""
+      SELECT
+        id,
+        name,
+        issue_date,
+        timezone_id,
+        created_on,
+        created_by,
+        created_email,
+        launched_on,
+        launched_by,
+        launched_email
+      FROM edition_issues
+      WHERE issue_date BETWEEN $dateFrom AND $dateTo AND name = ${edition.entryName}
+      """.map(EditionsIssue.fromRow(_)).list().apply()
+
+    maybeIssues.partitionMap(identity) match {
```

Comment:
This strikes me as a bit strange: I’m used to haskell, where `partition :: [Either a b] -> ([a], [b])`, so I had assumed this expression could be replaced by partition in scala as well. However, scala’s partition is instead roughly `partition: (List[A], A => Boolean) => (List[A], List[B])`, so I can’t see a nicer way to do this. I guess what we’re really looking for here is [Validated](https://typelevel.org/cats/datatypes/validated.html), maybe this would be nicer if using that. (Potentially a big change though!)

Reply:
I think leave it as-is here! I mostly mentioned it just to bring it to your attention :)